### PR TITLE
Monkey patch BasicSocket just for active_support < 4.1.0.

### PR DESCRIPTION
**Pull request by jondeandres**

### CODE REVIEW

**Comment by brianr** [comment]
:+1:

File: lib/rollbar.rb [hunk 1]
```diff
@@ -736,8 +736,19 @@ def require_core_extensions
...
       return if configuration.disable_core_monkey_patch
 
-      # Needed to avoid active_support bug serializing JSONs.
-      require 'rollbar/core_ext/basic_socket'
+      # Needed to avoid active_support (< 4.1.0) bug serializing JSONs
+      require 'rollbar/core_ext/basic_socket' if monkey_patch_socket?
+    end
+
+    def monkey_patch_socket?
+      return false unless defined(ActiveSupport::VERSION::STRING)
```
Comment (joneslee85) [inline lib/rollbar.rb]: what do you think about using `defined?`?

File: lib/rollbar.rb [hunk 2]
```diff
@@ -736,8 +736,19 @@ def require_core_extensions
...
+
+      major, minor = ActiveSupport::VERSION::STRING.split('.').map(&:to_i)
+
+      return true if major == 3
+      return true if major == 4 && minor == 0
```
Comment (joneslee85) [inline lib/rollbar.rb]: what do you think about consolidating 748 and 749 into 1 line?

File: lib/rollbar.rb [hunk 3]
```diff
@@ -736,8 +736,19 @@ def require_core_extensions
 
       return if configuration.disable_core_monkey_patch
 
-      # Needed to avoid active_support bug serializing JSONs.
-      require 'rollbar/core_ext/basic_socket'
+      # Needed to avoid active_support (< 4.1.0) bug serializing JSONs
+      require 'rollbar/core_ext/basic_socket' if monkey_patch_socket?
```
Comment (joneslee85) [inline lib/rollbar.rb]: What do you think about moving this under `rollbar/rails` directory?

File: lib/rollbar.rb (hunk 1 above)
Comment (jondeandres) [inline lib/rollbar.rb]: Oh yes, This should be `defined?` yes. Thanks!

File: lib/rollbar.rb (hunk 2 above)
Comment (jondeandres) [inline lib/rollbar.rb]: For me it's pretty clear in this way, thanks for the tip.

File: lib/rollbar.rb (hunk 3 above)
Comment (jondeandres) [inline lib/rollbar.rb]: While we don't refactor the gem, I'd prefer this code in this file. Thanks.

**Pull request merged**